lib/resourcemerge/core.go

```diff
@@ -56,6 +56,16 @@ func ensurePodSpec(modified *bool, existing *corev1.PodSpec, required corev1.Pod
 		}
 		ensureContainer(modified, existingCurr, required)
 	}
+	var requiredContainers []corev1.Container
+	for _, existing := range existing.Containers {
+		for _, required := range required.Containers {
+			if existing.Name == required.Name {
+				requiredContainers = append(requiredContainers, existing)
+				continue
+			}
+		}
+	}
+	existing.Containers = requiredContainers
```

Comment:
How does this (and the existing logic) square with [the PodSpec docs'][1]:

> Containers cannot currently be added or removed.

is there wrapping logic that removes the previous pod if `modified` and adds a new Pod with the new PodSpec?

[1]: https://github.com/openshift/cluster-version-operator/blob/40ba3b6bd45553b6118e250e8a0bdbc02e386ac5/vendor/k8s.io/api/core/v1/types.go#L2767

Reply:
A new pod will get created as part of deployment redeploy. So container will not be deleted from existing pod.